src/main/java/it/pagopa/pm/gateway/client/bpay/BancomatPayClient.java

```diff
@@ -1,33 +1,52 @@
 package it.pagopa.pm.gateway.client.bpay;
 
 import it.pagopa.pm.gateway.client.bpay.generated.*;
-import it.pagopa.pm.gateway.dto.*;
+import it.pagopa.pm.gateway.dto.BPayPaymentRequest;
+import it.pagopa.pm.gateway.dto.BPayRefundRequest;
 import it.pagopa.pm.gateway.utils.ClientUtils;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.ws.client.core.WebServiceTemplate;
 
-import javax.xml.bind.*;
+import javax.xml.bind.JAXBElement;
 import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static it.pagopa.pm.gateway.constant.ClientConfigs.*;
 
 @Slf4j
 public class BancomatPayClient {
 
     @Autowired
     private WebServiceTemplate webServiceTemplate;
 
-    @Value("${bancomatPay.client.group.code}")
-    public String GROUP_CODE;
-    @Value("${bancomatPay.client.institute.code}")
-    public String INSTITUTE_CODE;
-    @Value("${bancomatPay.client.tag}")
-    public String TAG;
-    @Value("${bancomatPay.client.token}")
-    public String TOKEN;
-
     private final ObjectFactory objectFactory = new ObjectFactory();
 
+    private final  Map<String, String> configValues;
+
+    {
+        configValues = getConfigValues();
+    }
+
+    @Value("${bancomatPay.client.config}")
+    private String BANCOMAT_CLIENT_CONFIG;
+
+    private Map<String, String> getConfigValues() {
+        List<String> listConfig = Arrays.asList(BANCOMAT_CLIENT_CONFIG.split(PIPE_SPLIT_CHAR));
```

Comment:
You need to check whether the environment variable you are reading is actually valued.
If the variable is null, then you will raise here a nullPointerException.
You can use the same logic used in PostePayPaymentTransactionController to read, check and split a custom environment property